File: app/src/main/java/com/gdi/hahi/mvp/contract/HahiContract.java
package com.gdi.hahi.mvp.contract;


import android.content.Context;

import com.gdi.hahi.base.BaseModel;
import com.gdi.hahi.base.BasePresenter;
import com.gdi.hahi.base.BaseView;

import java.util.Collection;

import io.reactivex.Observable;
import io.reactivex.Observer;

/**
 * Created by gandi on 2017/8/24 0024.
 */

public interface HahiContract {
    interface Model<T, V> extends BaseModel<T,V> {
        Observable<T> loadData(Context context,long lastTime);

    }

    interface View<T> extends BaseView {
        void setData(T data);
    }

    interface Presenter extends BasePresenter {
        void requestData(long lastTime);
    }


}

File: app/src/main/java/com/gdi/hahi/mvp/contract/HahiContract.kt
package com.gdi.hahi.mvp.contract


import android.content.Context

import com.gdi.hahi.base.BaseModel
import com.gdi.hahi.base.BasePresenter
import com.gdi.hahi.base.BaseView
import com.gdi.hahi.mvp.model.bean.NeiHanData

import io.reactivex.Observable

/**
 * Created by gandi on 2017/8/24 0024.
 */

interface HahiContract {
    interface Model<T, V> : BaseModel<T, V> {
        fun loadData(context: Context, lastTime: Long): Observable<T>

    }

    interface View<T> : BaseView {
        fun setData(data: T?)
    }

    interface Presenter : BasePresenter {
        fun requestData(lastTime: Long)
    }


}
